frontend/src/app/core/helpers/time-zone-helper.ts

```diff
@@ -1,18 +1,82 @@
-import { formatDate } from '@angular/common';
 import { ITimeZone } from '@core/models/ITimeZone';
-import { ILocalUser } from '@core/models/IUser';
+
+export const millisInHour = 3600 * 1000;
+export const millisInMinute = 60 * 1000;
+
+export const setTimeZone = (timeZone: ITimeZone) => {
+    localStorage.setItem('timeZone', JSON.stringify({ nameValue: timeZone.nameValue, timeValue: timeZone.timeValue }));
+};
+
+export const getDeviceTimeZone = () => {
+    const matches = new Date().toString().match(new RegExp('(?<=GMT)[\\+-]\\d{4}'));
+
+    if (matches === null) {
+        throw new Error('Cannot pick up current time zone');
+    }
+
+    const hours = matches[0].substring(0, 3);
```

Comment:
Maybe we can better use `new Date().getTimezoneOffset()` to get time zone device value?

Reply:
Rewrote using `getTimezoneOffset()`